### PR TITLE
Bench 276 sort integration tests for products and categories

### PR DESCRIPTION
**Pull request by JoeCSykes**

### CODE REVIEW

**Review by MichaelOSullivanAnswer — APPROVED** [review]

File: answerking_app/tests/test_integration/test_categories.py [hunk 1]
```diff
@@ -1,397 +1,298 @@
...
+@ddt()
+class GetTests(IntegrationTestBase):
+    def seed_cat_with_prod(self, cat_json, prod_json):
+        seeded_data_cat = self.seedFixture("categories", cat_json)
+        seeded_data_prod = self.seedFixture("products", prod_json)
+        cat_id = seeded_data_cat["id"]  # type: ignore[GeneralTypeIssue]
+        prod_id = seeded_data_prod["id"]  # type: ignore[GeneralTypeIssue]
+        cat: Category = Category.objects.get(pk=cat_id)
+        prod: Product = Product.objects.get(pk=prod_id)
+        cat.products.add(prod)
```
Comment (NogaAD) [inline answerking_app/tests/test_integration/test_categories.py]: you'll need to change this to `cat.product_set.add`

File: answerking_app/tests/test_integration/fixtures/products/basic-1-different-name.json [hunk 2]
```diff
@@ -3,6 +3,5 @@
     "name": "BurgerTwo",
     "description": "desc",
     "price": 1.2,
-    "category": [],
     "retired": false
-}
\ No newline at end of file
+}
```
Comment (NogaAD) [inline answerking_app/tests/test_integration/fixtures/products/basic-1-different-name.json]: Maybe add a product that uses 'categoryId'?

File: answerking_app/tests/test_integration/fixtures/products/basic-1-list.json [hunk 3]
```diff
@@ -4,8 +4,6 @@
         "name": "Burger",
         "price": "1.20",
         "description": "desc",
-        "stock": 100,
-        "calories": 100,
         "retired": false
```
Comment (NogaAD) [inline answerking_app/tests/test_integration/fixtures/products/basic-1-list.json]: I don't think you need 'retired' in the Json files, It set it as a default to false.

File: answerking_app/tests/test_integration/test_categories.py (hunk 1 above)
Comment (JoeCSykes) [reply answerking_app/tests/test_integration/test_categories.py:24]: changed

**Review by PietroConvalleAD — APPROVED** [review]

File: answerking_app/tests/test_integration/test_categories.py (hunk 1 above)
Comment (JoeCSykes) [reply answerking_app/tests/test_integration/test_categories.py]: Done

File: answerking_app/tests/test_integration/fixtures/products/basic-1-different-name.json (hunk 2 above)
Comment (JoeCSykes) [reply answerking_app/tests/test_integration/fixtures/products/basic-1-different-name.json]: Done

File: answerking_app/tests/test_integration/fixtures/products/basic-1-list.json (hunk 3 above)
Comment (JoeCSykes) [reply answerking_app/tests/test_integration/fixtures/products/basic-1-list.json]: Done

File: answerking_app/utils/mixins/CategoryProductMixins.py [hunk 4]
```diff
@@ -1,15 +1,25 @@
...
     def list(self, **kwargs) -> Response:
+        try:
+            Category.objects.get(pk=kwargs["pk"])
+        except ObjectDoesNotExist:
+            raise ProblemDetails(
+                status=status.HTTP_404_NOT_FOUND,
+                detail="Not Found",
+                title="Resource not found",
+            )
+
```
Comment (NogaAD) [inline answerking_app/utils/mixins/CategoryProductMixins.py:22]: Is this so we'll return 404 for a invalid category_id within a product request? instead the 400?

Comment (JoeCSykes) [reply answerking_app/utils/mixins/CategoryProductMixins.py:22]: no this is for the url "/api/categories/{id}/products" which returns a 404 if the category_id is not found  and should return a 400 if the id is not an integer.

**Review by NogaAD — APPROVED** [review]

**Pull request merged**